ast/term.go

```diff
@@ -1044,6 +1044,7 @@ type Set interface {
 	Map(func(*Term) (*Term, error)) (Set, error)
 	Reduce(*Term, func(*Term, *Term) (*Term, error)) (*Term, error)
 	Sorted() Array
+	Keys() []*Term
```

Comment:
This will be unneeded after the change to deal with empty sets that I mention below.

Reply:
Removed this